lib/dialects/abstract/connection-manager.js

```diff
@@ -243,6 +240,12 @@ class ConnectionManager {
   async getConnection(options) {
     options = options || {};
 
+    if (!this.pool) {
+      throw new Error(`
+        ConnectionManager.getConnection was called either after the pool was closed or before the pool was initialized`
+      );
+    }
```

Comment:
**Question:** Does it make sense to have a check on `initPools()` as follows?

```js
if (this.pool) {
  throw new Error("Cannot override initialized connection pool(s). Make sure the existing pool(s) are closed by first by invoking ConnectionManager.close");
}
```

This will prevent resource leakage by misusing the class.

Reply:
This is a good idea - let's see if @papb agrees and I'll add it.

I was also thinking of adding the following tests:
- Client should not be able to run `initPools()` if there already is an existing pool
- Client should be able to reuse same connectionManager to init pool, get connection, and close pool multiple times
- Client should not be able to call `getConnection` if pool has not been initialized
- Client should not be able to call `getConnection` if pool has been closed

Reply:
Hi @prestonlimlianjie and @marcogrcr, yes I agree!

@prestonlimlianjie please do it, and also add those tests, these are all great ideas. Thank you all very much. Sorry to take so long to give you the deserved attention.